src/doc/book/if-let.md

```diff
@@ -1,7 +1,8 @@
 % if let
 
-`if let` allows you to combine `if` and `let` together to reduce the overhead
-of certain kinds of pattern matches.
+`if let` allows us to match [patterns][patterns] within the condition of an [if][if].
+As a consequence, we reduce the overhead of certain kinds of [pattern][patterns] matches
```

Comment:
What about this?  I'd rather avoid using allow twice like that.  

`if let` permits pattern matching within the condition of an if statement.
This allows us to reduce the overhead of certain kinds of [pattern][patterns] matches
and express them in a more convenient way.